doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+
+Adaptive Routing System (ARS) allows dynamic selections of the best available path for data packets based on real-time network conditions. This approach helps to mitigate congestion, optimize resource utilization, and improve overall network performance.
+
+No standard exists, but there is an industry consensus for general approach.
+
+ARS can be divided into two parts:
+- Local ARS - Scope of this HLD, focuses on dynamic path selection within a single device. It enables more granular control over routing decisions by considering the real-time utilization of local egress ports. This is achieved by monitoring local port utilization, identifying micro flows inside of incoming macro flow and assigning best available path.
+- Global ARS - focuses on selecting non-congested path to end host by exchanging ports utilization data, between remote devices.
+
+This design follows SAI conceptual model described in https://github.com/opencomputeproject/SAI/blob/master/doc/ARS/Adaptive-Routing-and-Switching.md.
+
+
+The diagram illustrates a Conceptual Packet Flow, where macro flows are collections of multiple micro flows. Micro flows, identified by a 5-tuple (source IP, destination IP, protocol, source port, destination port), are hashed, and if several micro flows hash to the same bucket, they are grouped into a macro flow. These flows are further segmented into flow-lets based on idle time thresholds between packets. A Macro Flow Table maps these flows to next-hop destinations and tracks their status (e.g., "active" or "expired"). Using Adaptive Routing within an ECMP group, the system dynamically assigns traffic based on link quality metrics like latency and packet loss.
+
+__Figure 1: ARS Packet Flow__
+![ARS flow](images/ARSPacketFlow.png "Figure 1: ARS Packet Flow")
+This diagram outlines a Logical Pipeline Flow for Adaptive Routing with ARS. Flow starts with L3 Route Lookup to identify the appropriate Next-Hop Group (NHG). If ARS is enabled, the system generates an ARS Macro Flow ID and assigns packets to specific next-hop destinations using a Macro Flow Table. Otherwise, traffic undergoes standard ECMP processing. In the background, ARS monitors port load metrics (past and future) and updates the Macro Flow Table using adaptive algorithms and quality thresholds to ensure optimal traffic distribution based on real-time link conditions.
+
+__Figure 2: ARS SAI Pipeline Flow__
+![SAI model](images/ARSPipeline.png "Figure 2: ARS Pipeline Flow")
+
+### Use cases
+- L3 traffic egressing via NHG
+    * NH pointing to port
+    * NH pointing to lag
+- L2 traffic egressing via LAG
+- Tunnel interface as egress interface is not supported
+
+### Requirements
+
+* Phase 1
+1. Support different ARS modes:
+    - Flowlet-based port selection
+    - Per packet port selection
+2. Support enabling ARS over NHG
+3. Support path quality configuration

```

Comment:
Please make sure that the design supports 2 mode of operation regarding creation of NHG: done by Route OA and done by FRR when it decides on creation and push the NHG to APP DB